### PR TITLE
[#860] Render Generic Template Carousel from…

### LINKED ISSUE #860: Render Generic Template Carousel from Facebook
```
{
   "conversation_id": "UUID",
   "message":{
     "attachment":{
      "type":"template",
      "payload":{
        "template_type":"generic",
        "elements":[
           {
            "title":"Welcome!",
            "image_url":"https://petersfancybrownhats.com/company_image.png",
            "subtitle":"We have the right hat for everyone.",
            "default_action": {
              "type": "web_url",
              "url": "https://petersfancybrownhats.com/view?item=103",
              "webview_height_ratio": "tall",
            },
            "buttons":[
              {
                "type":"web_url",
                "url":"https://petersfancybrownhats.com",
                "title":"View Website"
              },{
                "type":"postback",
                "title":"Start Chatting",
                "payload":"DEVELOPER_DEFINED_PAYLOAD"
              }              
            ]      
          },
          {
            "title":"Card 2",
            "image_url":"https://petersfancybrownhats.com/company_image.png",
            "subtitle":"Subtitle Card 2",
            "default_action": {
              "type": "web_url",
              "url": "https://petersfancybrownhats.com/view?item=103",
              "webview_height_ratio": "tall",
            },
            "buttons":[
              {
                "type":"web_url",
                "url":"https://petersfancybrownhats.com",
                "title":"View Website"
              },{
                "type":"postback",
                "title":"Start Chatting",
                "payload":"DEVELOPER_DEFINED_PAYLOAD"
              }              
            ]      
          }
        ]
      }
    }
  }
}
```

### PR DESCRIPTION
**Pull request by bitboxer**
…Facebook

closes #860

### CODE REVIEW

File: lib/typescript/render/assets/icons/chevron_left.svg [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
```
Comment (bitboxer) [inline lib/typescript/render/assets/icons/chevron_left.svg:1]: I know that this duplicates the svg to the render library but I didn't want to create a real asset library and move all svg to that inside of this PR. We need another ticket to do that 😉

**Comment by bitboxer** [comment]
The arrow buttons are only visible if scrolling is possible like here:

Screen Shot 2021-02-11 at 15 11 07

They are hidden if not needed like here: 

Screen Shot 2021-02-11 at 15 11 11

Also if you resize the browser window the buttons show and hide depending on the width of the view.

File: lib/typescript/render/providers/facebook/components/GenericTemplate/index.tsx [hunk 2]
```diff
@@ -1,36 +1,91 @@
...
+      if (templatesEl.current.scrollWidth > templatesEl.current.clientWidth) {
+        buttonEl.current.style.display = 'block';
+      } else {
+        console.log('hide button');
+        buttonEl.current.style.display = 'none';
+      }
+    }
+  }, [templatesEl, buttonEl]);
+
+  const registerObeserver = useCallback(() => {
```
Comment (AitorAlgorta) [inline lib/typescript/render/providers/facebook/components/GenericTemplate/index.tsx:40]: Obese server 😄

File: lib/typescript/render/providers/facebook/components/GenericTemplate/index.tsx [hunk 3]
```diff
@@ -1,36 +1,91 @@
...
+      behavior: 'smooth',
+    });
+  }, [templatesEl]);
+
+  const resetScrollButtons = useCallback(() => {
+    if (buttonEl.current) {
+      if (templatesEl.current.scrollWidth > templatesEl.current.clientWidth) {
+        buttonEl.current.style.display = 'block';
+      } else {
+        console.log('hide button');
```
Comment (AitorAlgorta) [inline lib/typescript/render/providers/facebook/components/GenericTemplate/index.tsx:34]: 💣

**Review by AitorAlgorta — APPROVED** [review]
LGTM, small comment

**Review by AitorAlgorta — APPROVED** [review]
🏅

**Pull request merged**